Provisioning: fix multi-tenant and single-tenant authorization

MissingRoberto opened the pull request:
## Summary

This PR adds an `ExtraAudience` option to the `RoundTripper` and refactors the authorization system for provisioning resources with a centralized access checker that handles fine-grained permissions and role-based fallback.

## Context & Issues Addressed

This refactor addresses several authorization inconsistencies in the provisioning API:

- **Incorrect audience for MT API**: The provisioning group audience was being added incorrectly for the multi-tenant Provisioning API server itself, when it should only be used by operators calling other API servers through the aggregator.
- **Inconsistent fallback behavior**: Role-based fallbacks were missing in some code paths, causing authorization failures for users who should have had access via their org role.
- **Mixed authorization modes**: The `useExclusivelyAccessCheckerForAuthz` flag was not consistently enforced across all endpoints, leading to unpredictable behavior between fine-grained and role-based authorization.
- **Incorrect fallback roles**: Some endpoints were using admin-level fallback when editor-level would be more appropriate (e.g., job management).
- **Missing fine-grained permissions**: Several endpoints defaulted to role-based admin checks without proper fine-grained permission support.
- **New endpoints without permissions**: The connections endpoints were missing fine-grained permission definitions entirely.
- **Missing RBAC mapper entries**: Settings, stats, and connections resources were not mapped in the authz RBAC mapper, causing MT authorization failures.
- **Verb mismatch for custom endpoints**: Settings and stats are custom routes (not K8s resources). The K8s apiserver interprets requests to collection endpoints like `/settings` as "list" operations, but we were hardcoding `VerbGet` which caused mismatches.

## Double Authorization Challenge (ST → MT)

When a user makes a request to provisioning APIs in a multi-tenant setup, authorization happens **twice**:

```
User Request
    ↓
┌─────────────────────────────────────────────┐
│ ST Grafana (Single Tenant)                  │
│  ├─ User session with OrgRole from DB      │
│  ├─ Authorizer runs (GetAuthorizer)        │
│  └─ Uses sessionAccessChecker with fallback│
└─────────────────────────────────────────────┘
    ↓ (via aggregator, X-Access-Token + X-Grafana-Id headers)
┌─────────────────────────────────────────────┐
│ MT API Server (Multi Tenant)                │
│  ├─ Extended JWT authentication            │
│  │   └─ OrgRole NOT populated (no DB)      │
│  ├─ Authorizer runs again                  │
│  └─ Uses tokenAccessChecker (no fallback)  │
│       └─ Calls IAM authz service           │
└─────────────────────────────────────────────┘
```

### The Problem

1. **ST Grafana**: User has session with `OrgRole` populated from database. Authorization passes via role-based fallback.

2. **MT API Server**: Extended JWT creates identity WITHOUT `OrgRoles` (no access to Grafana DB). The IAM authz service must have:
   - Fine-grained permissions registered (`accesscontrol.go`)
   - RBAC mapper entries to translate K8s verbs → legacy actions (`mapper.go`)

Without both, the MT authorizer returns "permission denied" even for valid users.

### Custom Endpoints vs K8s Resources

Settings and stats are **custom routes** defined via `GetAPIRoutes()`, not standard Kubernetes resources. When the K8s apiserver receives a GET request to `/settings` (without a specific resource name), it interprets this as a LIST operation:

- `GET /settings` → verb = `list` (collection endpoint)
- `GET /settings/my-setting` → verb = `get` (specific resource)

The authorization must use `a.GetVerb()` to get the actual verb from the request, rather than hardcoding `VerbGet`. The RBAC mapper translates both `list` and `get` to `:read` actions.

### The Solution

1. **Fine-grained permissions**: Added `provisioning.settings:read` and `provisioning.stats:read` actions
2. **Role grants**: Settings granted to Viewer, Stats granted to Admin
3. **RBAC mapper**: Added translations for `connections`, `settings`, `stats` resources
4. **Access checker**: Uses IAM authz service which checks registered permissions
5. **Use actual verb**: Use `a.GetVerb()` instead of hardcoded verbs for authorization

## Changes

### RoundTripper (`round_tripper.go`)
- Added `ExtraAudience(audience string)` `RoundTripperOption` that appends an additional audience to token exchange requests
- Added forwarding of user ID token via `X-Grafana-Id` header for aggregator forwarding
- This allows operators to request tokens valid for both the target API server and the provisioning group
- Improved documentation and comments

### New Centralized Auth Package (`apps/provisioning/pkg/auth`)

Created a new `AccessChecker` interface with two implementations based on identity source:

#### `TokenAccessChecker` (`token_access_checker.go`)
- Gets identity from access tokens via `AuthInfoFrom(ctx)`
- No role-based fallback supported (MT has no access to user DB)
- `WithFallbackRole()` is a no-op (returns same instance)
- Debug logging for authorization troubleshooting

#### `SessionAccessChecker` (`session_access_checker.go`)
- Gets identity from Grafana sessions via `GetRequester(ctx)`
- Supports optional role-based fallback via `WithFallbackRole()` for backwards compatibility
- Debug logging for authorization troubleshooting

#### Common Features
- **Automatic namespace handling**: Fills `req.Namespace` from identity if not provided
- **API error returns**: Returns `apierrors.Forbidden` or `apierrors.NewUnauthorized` directly
- **AccessPolicy passthrough**: Trusts AccessPolicy identities as internal callers
- **Mockery-generated mock**: `MockAccessChecker` for testing
- **Debug logging**: Uses grafana-app-sdk logger for detailed authorization tracing

```go
type AccessChecker interface {
    Check(ctx context.Context, req authlib.CheckRequest, folder string) error
    WithFallbackRole(role identity.RoleType) AccessChecker
}
```

### Fine-Grained Permissions for Settings and Stats (`accesscontrol.go`)

Added new fine-grained permissions for read-only endpoints:
- `provisioning.settings:read` - granted to Viewer role (all logged-in users)
- `provisioning.stats:read` - granted to Admin role only

Added role registrations:
- `settingsReader` role granted to `org.RoleViewer`
- `statsReader` role granted to `org.RoleAdmin`

### RBAC Mapper Updates (`mapper.go`)

Added missing resource translations for the authz service:
- `connections` → `provisioning.connections`
- `settings` → `provisioning.settings`
- `stats` → `provisioning.stats`

Without these mappings, the authz service could not translate Kubernetes verbs to legacy actions, causing 403 errors in MT mode.

### Authorization Refactor (`register.go`, `files.go`, `dualwriter.go`)

- Replaced scattered authorization logic with the new centralized `auth.AccessChecker`
- Uses `accessWithAdmin`, `accessWithEditor`, and `accessWithViewer` instances
- Simplified call sites with `toAuthorizerDecision()` helper for route-level authorization
- Removed role-based authorization functions (`isRoleBasedResource`, `authorizeRoleBasedResource`)
- **Fixed verb handling**: Use `a.GetVerb()` for all resources to get the actual request verb, not hardcoded values

#### Resource Authorization
| Resource | Fine-grained Permission | Fallback Role (ST only) |
|----------|------------------------|-------------------------|
| Repositories | `provisioning.repositories:*` | Admin |
| Connections | `provisioning.connections:*` | Admin |
| Jobs | `provisioning.jobs:*` | **Editor** |
| HistoricJobs | `provisioning.historicjobs:read` | Admin |
| Settings | `provisioning.settings:read` | Viewer |
| Stats | `provisioning.stats:read` | Admin |

#### Files Subresource
- Route level allows any authenticated user
- Directory listing: checks `repositories:read` in connector
- Individual file CRUD: delegated to `DualReadWriter` which checks actual resource permissions

#### Subresource Verb Handling
For subresources with fixed access patterns, verbs are intentionally hardcoded:
- `test` → `VerbUpdate` (write-like operation)
- `refs`, `resources`, `history`, `status` → `VerbGet` (read-only subresources)

For full resources (repositories, connections, jobs, historicjobs, settings, stats), `a.GetVerb()` is used.

## Motivation

### ExtraAudience Option
Operators connecting to the multitenant aggregator need tokens that include both:
1. The target API server's audience (e.g., `dashboards.grafana.app`, `folders.grafana.app`)
2. The provisioning group audience (`provisioning.grafana.app`)

**Important**: This option is for **operators only** when making cross-API calls through the aggregator. The MT Provisioning API server itself should not include additional audiences in its own token requests.

### Centralized Access Checker
- **Identity-source based implementations**: Names reflect how identity is obtained
- **Consistent fallback behavior**: Role fallback only in session mode for backwards compatibility
- **Simpler call sites**: Returns errors directly, handles namespace automatically
- **Testable**: Full unit test coverage with separate test files per implementation
- **Debug logging**: Comprehensive logging for troubleshooting authorization issues

## Testing

- Unit tests for `TokenAccessChecker` and `SessionAccessChecker` in separate test files
- Mockery-generated `MockAccessChecker` for use in other tests
- Integration tests pass for connection and repository CRUD operations
- All existing tests pass

## Related

Enterprise PR: grafana/grafana-enterprise#10448

MissingRoberto committed:
feat(auth): add ExtraAudience option to RoundTripper

Add ExtraAudience option to RoundTripper to allow operators to include
additional audiences (e.g., provisioning group) when connecting to the
multitenant aggregator. This ensures tokens include both the target API
server's audience and the provisioning group audience, which is required
to pass the enforceManagerProperties check.

- Add ExtraAudience RoundTripperOption
- Improve documentation and comments
- Add comprehensive test coverage

MissingRoberto committed:
fix(operators): add ExtraAudience for dashboards/folders API servers

Operators connecting to dashboards and folders API servers need to include
the provisioning group audience in addition to the target API server's
audience to pass the enforceManagerProperties check.

MissingRoberto committed:
Merge remote-tracking branch 'origin/main' into feat/round-tripper-extra-audience-option

MissingRoberto committed:
provisioning: fix settings/stats authorization for AccessPolicy identities

The settings and stats endpoints were returning 403 for users accessing via
ST->MT because the AccessPolicy identity was routed to the access checker,
which doesn't know about these resources.

This fix handles 'settings' and 'stats' resources before the access checker
path, routing them to the role-based authorization that allows:
- settings: Viewer role (read-only, needed by frontend)
- stats: Admin role (can leak information)

MissingRoberto committed:
fix: update BootstrapStep component to remove legacy storage handling and adjust resource counting logic

- Removed legacy storage flag from useResourceStats hook in BootstrapStep.
- Updated BootstrapStepResourceCounting to simplify rendering logic and removed target prop.
- Adjusted tests to reflect changes in resource counting and rendering behavior.

MissingRoberto committed:
Revert "fix: update BootstrapStep component to remove legacy storage handling and adjust resource counting logic"

This reverts commit 148802cbb59722d48347f09ff7351dc747f272a6.

MissingRoberto committed:
provisioning: allow any authenticated user for settings/stats endpoints

These are read-only endpoints needed by the frontend:
- settings: returns available repository types and configuration for the wizard
- stats: returns resource counts

Authentication is verified before reaching authorization, so any user who
reaches these endpoints is already authenticated. Requiring specific org
roles failed for AccessPolicy tokens which don't carry traditional roles.

MissingRoberto committed:
provisioning: remove redundant admin role check from listFolderFiles

The admin role check in listFolderFiles was redundant (route-level auth already
handles access) and broken for AccessPolicy identities which don't have org roles.

File access is controlled by the AccessClient as documented in the route-level
authorization comment.

MissingRoberto committed:
Merge remote-tracking branch 'origin/main' into feat/round-tripper-extra-audience-option

MissingRoberto committed:
provisioning: add isAdminOrAccessPolicy helper for auth checks

Consolidates authorization logic for provisioning endpoints:
- Adds isAdminOrAccessPolicy() helper that allows admin users OR AccessPolicy identities
- AccessPolicy identities (ST->MT flow) are trusted internal callers without org roles
- Regular users must have admin role (matching frontend navtree restriction)

Used in: authorizeSettings, authorizeStats, authorizeJobs, listFolderFiles

MissingRoberto committed:
provisioning: consolidate auth helpers into allowForAdminsOrAccessPolicy

Simplifies authorization by:
- Adding isAccessPolicy() helper for AccessPolicy identity check
- Adding allowForAdminsOrAccessPolicy() that returns Decision directly
- Consolidating stats/settings/jobs into single switch case
- Using consistent pattern in files.go

MissingRoberto committed:
provisioning: require admin for files subresource at route level

Aligns route-level authorization with handler-level check in listFolderFiles.
Both now require admin role OR AccessPolicy identity for consistency.

MissingRoberto committed:
provisioning: restructure authorization with role-based helpers

Reorganizes authorization code for clarity:

Role-based helpers (all support AccessPolicy for ST->MT flow):
- allowForAdminsOrAccessPolicy: admin role required
- allowForEditorsOrAccessPolicy: editor role required
- allowForViewersOrAccessPolicy: viewer role required

Repository subresources by role:
- Admin: repository CRUD, test, files
- Editor: jobs, resources, sync, history
- Viewer: refs, status (GET only)

Connection subresources by role:
- Admin: connection CRUD
- Viewer: status (GET only)

MissingRoberto committed:
provisioning: move refs to admin-only

refs subresource now requires admin role (or AccessPolicy).
Updated documentation comments to reflect current permissions.

MissingRoberto committed:
provisioning: add fine-grained permissions for connections

Adds connection permissions following the same pattern as repositories:
- provisioning.connections:create
- provisioning.connections:read
- provisioning.connections:write
- provisioning.connections:delete

Roles:
- fixed:provisioning.connections:reader (granted to Admin)
- fixed:provisioning.connections:writer (granted to Admin)

MissingRoberto committed:
provisioning: remove non-existent sync subresource from auth

The sync subresource doesn't exist - syncing is done via the jobs endpoint.
Removed dead code from authorization switch case.

MissingRoberto committed:
provisioning: use access checker for fine-grained permissions

Refactors authorization to use b.access.Check() with verb-based checks:

Repository subresources:
- CRUD: uses actual verb (get/create/update/delete)
- test: uses 'update' (write permission)
- files/refs/resources/history/status: uses 'get' (read permission)
- jobs: uses actual verb for jobs resource

Connection subresources:
- CRUD: uses actual verb
- status: uses 'get' (read permission)

The access checker maps verbs to actions defined in accesscontrol.go.
Falls back to admin role for backwards compatibility.

Also removes redundant admin check from listFolderFiles since
authorization is now properly handled at route level.

MissingRoberto committed:
provisioning: use verb constants instead of string literals

Uses apiutils.VerbGet, apiutils.VerbUpdate instead of "get", "update".

MissingRoberto committed:
provisioning: use access checker for jobs and historicjobs resources

Jobs resource: uses actual verb (create/read/write/delete)
HistoricJobs resource: read-only (historicjobs:read)

MissingRoberto committed:
provisioning: allow viewers to access settings endpoint

Settings is read-only and needed by multiple UI pages (not just admin pages).
Stats remains admin-only.

MissingRoberto committed:
provisioning: consolidate role-based resource authorization

Extract isRoleBasedResource() and authorizeRoleBasedResource() helpers
to avoid duplicating settings/stats resource checks in multiple places.

MissingRoberto committed:
provisioning: use resource name constants instead of hardcoded strings

Replace 'repositories', 'connections', 'jobs', 'historicjobs' with
their corresponding ResourceInfo.GetName() constants.

MissingRoberto committed:
provisioning: delegate file authorization to connector

Route level: allow any authenticated user for files subresource
Connector: check repositories:read only for directory listing
Individual file CRUD: handled by DualReadWriter based on actual resource

MissingRoberto committed:
provisioning: enhance authorization for files and jobs resources

Updated file authorization to fall back to admin role for listing files. Introduced checkAccessForJobs function to manage job permissions, allowing editors to create and manage jobs while maintaining admin-only access for historic jobs. Improved error messaging for permission denials.

MissingRoberto committed:
provisioning: refactor authorization with fine-grained permissions

Authorization changes:
- Use access checker with role-based fallback for backwards compatibility
- Repositories/Connections: admin role fallback
- Jobs: editor role fallback (editors can manage jobs)
- HistoricJobs: admin role fallback (read-only)
- Settings: viewer role (needed by multiple UI pages)
- Stats: admin role

Files subresource:
- Route level allows any authenticated user
- Directory listing checks repositories:read in connector
- Individual file CRUD delegated to DualReadWriter

Refactored checkAccessWithFallback to accept fallback role parameter.

MissingRoberto committed:
provisioning: refactor access checker integration for improved authorization

Updated the authorization logic to utilize the new access checker across various resources, including files and jobs. This change simplifies the permission checks by removing redundant identity retrieval and enhances error handling. The access checker now supports role-based fallbacks for admin and editor roles, ensuring backward compatibility while streamlining the authorization process for repository and connection subresources.

MissingRoberto committed:
provisioning: remove legacy access checker tests and refactor access checker implementation

Deleted the access_checker_test.go file to streamline the codebase and focus on the updated access checker implementation. Refactored the access checker to enhance clarity and maintainability, ensuring it supports role-based fallback behavior. Updated the access checker integration in the API builder to utilize the new fallback role configuration, improving authorization logic across resources.

MissingRoberto committed:
refactor: split AccessChecker into TokenAccessChecker and SessionAccessChecker

- Renamed NewMultiTenantAccessChecker -> NewTokenAccessChecker (uses AuthInfoFrom)
- Renamed NewSingleTenantAccessChecker -> NewSessionAccessChecker (uses GetRequester)
- Split into separate files with their own tests
- Added mockery-generated mock for AccessChecker interface
- Names now reflect identity source rather than deployment mode

MissingRoberto commented:
/enterprise

MissingRoberto committed:
fix: correct error message case and use accessWithAdmin for filesConnector

- Fixed error message to use lowercase 'admin role is required'
- Fixed filesConnector to use accessWithAdmin for proper role fallback
- Formatted code

charandas commented on apps/provisioning/pkg/auth/token_access_checker.go:
```diff
@@ -0,0 +1,62 @@
+	}
+
+	// AccessPolicy identities are trusted internal callers
+	if authlib.IsIdentityType(id.GetIdentityType(), authlib.TypeAccessPolicy) {
```
Is this always the case? I don't think so when this code is running in the MT Service.

In ST service, we have abused TypeAccessPolicy in lieu of Service Identity which was supposed to be used in-process only.

When we pepper our code with such checks, its hard to reason whether the identity type in this case is coming in from the wire, or from within process. If its coming from the wire, then we need to ensure adequate permissions here or somewhere down the chain).

On another note, we should extend the permissions [here](https://github.com/grafana/grafana/blob/f3d4181cf27be01384654af6645d14e87e5afd42/pkg/apimachinery/identity/context.go#L129-L153) and actually verify that these permissions hold, instead of trusting them blindly.

MissingRoberto replied:
yea, I changed the behavior. Well spotted.

MissingRoberto replied:
> On another note, we should extend the permissions [here](https://github.com/grafana/grafana/blob/f3d4181cf27be01384654af6645d14e87e5afd42/pkg/apimachinery/identity/context.go#L129-L153) and actually verify that these permissions hold, instead of trusting them blindly.

Aren't those already validated if we call `access.Check` for the context? How would that work? @charandas

MissingRoberto committed:
refactor: reduce cyclomatic complexity in filesConnector.Connect

Split the Connect handler into smaller focused functions:
- handleRequest: main request processing
- createDualReadWriter: setup dependencies
- parseRequestOptions: extract request options
- handleDirectoryListing: GET directory requests
- handleMethodRequest: route to method handlers
- handleGet/handlePost/handlePut/handleDelete: method-specific logic
- handleMove: move operation logic

MissingRoberto committed:
security: remove blind TypeAccessPolicy bypass from access checkers

Removed the code that bypassed authorization for TypeAccessPolicy identities.
All identities now go through proper permission verification via the inner
access checker, which will validate permissions from ServiceIdentityClaims.

This addresses the security concern where TypeAccessPolicy was being trusted
blindly without verifying whether the identity came from the wire or in-process.

MissingRoberto committed:
feat: allow editors to access repository refs subresource

Change refs authorization from admin to editor fallback so editors can
view repository branches when pushing changes to dashboards/folders.

- Split refs from other read-only subresources (resources, history, status)
- refs now uses accessWithEditor instead of accessWithAdmin
- Updated documentation comment to reflect authorization levels
- Added integration test TestIntegrationProvisioning_RefsPermissions
  verifying editor access and viewer denial

MissingRoberto committed:
tests: add authorization tests for missing provisioning API endpoints

Add comprehensive authorization tests for:
- Repository subresources (test, resources, history, status)
- Connection status subresource
- HistoricJobs resource
- Settings and Stats resources

All authorization paths are now covered by integration tests.

MissingRoberto committed:
test: fix RefsPermissions test to use GitHub repository

Use github-readonly.json.tmpl template instead of local folder,
since refs endpoint requires a versioned repository that supports
git operations.

MissingRoberto committed:
chore: format test files

MissingRoberto committed:
fix: make settings/stats authorization work in MT mode

Update authorizeRoleBasedResource to check authlib.AuthInfoFrom(ctx)
for AccessPolicy identity type in addition to identity.GetRequester(ctx).
This ensures AccessPolicy identities are recognized in MT mode where
identity.GetRequester may not set the identity type correctly.

MissingRoberto committed:
fix: remove unused authorization helper functions

Remove allowForAdminsOrAccessPolicy and allowForViewersOrAccessPolicy
as they are no longer used after refactoring to use authorizeRoleBasedResource.

MissingRoberto committed:
Fix AccessPolicy identity detection in ST authorizer

- Add check for AccessPolicy identities via GetAuthID() in authorizeRoleBasedResource
- Extended JWT may set identity type to TypeUser but AuthID is 'access-policy:...'
- Forward user ID token in X-Grafana-Id header in RoundTripper for aggregator forwarding

MissingRoberto committed:
Revert "Fix AccessPolicy identity detection in ST authorizer"

This reverts commit 0f4885e503a633d5e78252842d8885d396ad1bab.

MissingRoberto committed:
Merge remote-tracking branch 'origin/main' into feat/round-tripper-extra-audience-option

MissingRoberto committed:
Add fine-grained permissions for settings and stats endpoints

- Add provisioning.settings:read action (granted to Viewer role)
- Add provisioning.stats:read action (granted to Admin role)
- Add accessWithViewer to APIBuilder for Viewer role fallback
- Use access checker for settings/stats authorization
- Remove role-based authorization functions (isRoleBasedResource, authorizeRoleBasedResource)

This makes settings and stats consistent with other provisioning resources
and works properly in both ST and MT modes via the access checker.

MissingRoberto committed:
Remove AUTHORIZATION_COVERAGE.md

MissingRoberto committed:
Add provisioning resources to RBAC mapper

- Add connections, settings, stats to provisioning.grafana.app mappings
- Required for authz service to translate K8s verbs to legacy actions
- Fixes 403 errors for settings/stats in MT mode

MissingRoberto commented:
/deploy-to-hg

MissingRoberto commented:
/deploy-to-hg

MissingRoberto commented:
/deploy-to-hg

MissingRoberto committed:
Merge remote-tracking branch 'origin/main' into feat/round-tripper-extra-audience-option

MissingRoberto commented:
/deploy-to-hg

MissingRoberto committed:
refactor: merge access checkers with original fallthrough behavior

Merge tokenAccessChecker and sessionAccessChecker into a unified
access checker that implements the original fallthrough behavior:

1. First try to get identity from access token (authlib.AuthInfoFrom)
2. If token exists AND (is TypeAccessPolicy OR useExclusivelyAccessCheckerForAuthz),
   use the access checker with token identity
3. If no token or conditions not met, fall back to session identity
   (identity.GetRequester) with optional role-based fallback

This fixes the issue where settings/stats/connections endpoints were
failing in MT mode because the tokenAccessChecker was returning an error
when there was no auth info in context, instead of falling through to
session-based authorization.

The unified checker now properly handles:
- MT mode: tries token first, falls back to session if no token
- ST mode: only uses token for AccessPolicy identities, otherwise session
- Role fallback: applies when configured and access checker denies

charandas committed:
Revert "refactor: merge access checkers with original fallthrough behavior"

This reverts commit 96451f948bf8d723845567d2a26e7ecffda483ba.

charandas committed:
Grant settings view role to all

charandas commented:
/deploy-to-hg --enterprise-ref feat/round-tripper-extra-audience-option

MissingRoberto committed:
fix: use actual request verb for settings/stats authorization

Use a.GetVerb() instead of hardcoded VerbGet for settings and stats
authorization. When listing resources (hitting collection endpoint),
the verb is 'list' not 'get', and this mismatch could cause issues
with the RBAC service.

MissingRoberto committed:
debug: add logging to access checkers for authorization debugging

Add klog debug logs (V4 level) to token and session access checkers
to help diagnose why settings/stats authorization is failing while
connections works.

MissingRoberto committed:
debug: improve access checker logging with grafana-app-sdk logger

- Use grafana-app-sdk logging.FromContext instead of klog
- Add error wrapping with resource.group format for better context
- Log more details including folder, group, and allowed status
- Log error.Error() for better error message visibility

MissingRoberto committed:
chore: use generic log messages in access checkers

MissingRoberto committed:
Merge remote-tracking branch 'origin/main' into feat/round-tripper-extra-audience-option

MissingRoberto commented on pkg/registry/apis/provisioning/accesscontrol.go:
```diff
@@ -119,11 +171,47 @@ func registerAccessControlRoles(service accesscontrol.Service) error {
+				},
+			},
+		},
+		Grants: []string{string(org.RoleViewer), string(org.RoleEditor), string(org.RoleAdmin)},
```
I guess you meant `RoleAdmin` :)

MissingRoberto committed:
Revert "Grant settings view role to all"

This reverts commit 3f5758cf3656b7e6c267e7a3e83d98da1949ab20.

MissingRoberto committed:
Merge remote-tracking branch 'origin/main' into feat/round-tripper-extra-audience-option

MissingRoberto committed:
fix: use request verb for historicjobs authorization

The original role-based check allowed any verb for admins. To preserve
this behavior with the access checker, we should pass the actual verb
from the request instead of hardcoding VerbGet.

ferruvich reviewed (approved):
LGTM!

MissingRoberto committed:
Merge remote-tracking branch 'origin/main' into feat/round-tripper-extra-audience-option

MissingRoberto committed:
Merge remote-tracking branch 'origin/main' into feat/round-tripper-extra-audience-option
